inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api.pyx

```diff
@@ -245,48 +245,37 @@ cdef class IECore:
         return versions
 
     ## Reads a network from the Intermediate Representation (IR) and creates an `IENetwork`.
-    #  @param model: A `.xml`, `.onnx`or `.prototxt` model file or string with IR.
-    #  @param weights: A `.bin` file of the IR. Depending on `init_from_buffer` value, can be a string path or
-    #                  bytes with file content.
-    #  @param init_from_buffer: Defines the way of how `model` and `weights` attributes are interpreted.
-    #                           If  `False`, attributes are interpreted as strings with paths to .xml and .bin files
-    #                           of IR. If `True`, they are  interpreted as Python `bytes` object with .xml and .bin files content.
+    #  @param model: A `.xml`, `.onnx`or `.prototxt` model file or a 'bytes' object which contains a model.
+    #  @param weights: A `.bin` file of the IR. Can be a string path or bytes with file content.
     #  @return An `IENetwork` object
     #
     #  Usage example:\n
     #  ```python
     #  ie = IECore()
     #  net = ie.read_network(model=path_to_xml_file, weights=path_to_bin_file)
     #  ```
-    cpdef IENetwork read_network(self, model: [str, bytes, Path], weights: [str, bytes, Path] = "", init_from_buffer: bool = False):
-        cdef uint8_t*bin_buffer
-        cdef string weights_
-        cdef string model_
+    cpdef IENetwork read_network(self, model : [str, bytes, Path], weights : [str, bytes, Path] = ""):
         cdef IENetwork net = IENetwork()
-        if init_from_buffer:
+        cdef uint8_t*bin_buffer
+        if isinstance(model, Path):
+            if not model.is_file():
+                raise Exception("Path to the model {} doesn't exist or it's a directory".format(model))
+            model_ = bytes(model)
+            if isinstance(weights, Path) :
+                weights_ = bytes(weights)
```

Comment:
I think here check of weights is needed. If type of weights is `Path`, it means that user specify weights, so they must be valid.